presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoConfigUtils.java

```diff
@@ -74,4 +74,123 @@ public static void deployConfigFiles(File configTar, Path configDir)
             throw new PrestoManagerException("Failed to deploy config files", untarResult);
         }
     }
+
+    public static void addConfigFiles(Path configDir)
+            throws PrestoManagerException
+    {
+        try {
+            deleteDirectoryContents(configDir);
+            if (containsRegularFiles(DEFAULT_CONFIG_DIR)) {
+                copyDir(DEFAULT_CONFIG_DIR, configDir);
+                updateProperty(configDir.resolve("node.properties"), "node.id", randomUUID().toString());
+                addEnvSh(configDir);
+            }
+            else {
+                addWorkerConfig(configDir);
+            }
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to add config files", e);
+        }
+    }
+
+    private static void addEnvSh(Path configDir)
+            throws PrestoManagerException
+    {
+        Properties properties = new Properties();
+        properties.setProperty("JAVA8_HOME", System.getProperty("java.home"));
+        createPropertiesFile(properties, configDir.resolve("env.sh"), null);
+    }
+
+    private static void createPropertiesFile(Properties properties, Path path, String comments)
+            throws PrestoManagerException
+    {
+        try (OutputStream output = newOutputStream(path, CREATE_NEW)) {
+            properties.store(output, comments);
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException(format("Failed to create properties file: %s", path.getFileName()), e);
+        }
+    }
+
+    public static void addConnectors(Path catalogDir)
+            throws PrestoManagerException
+    {
+        try {
+            if (containsRegularFiles(DEFAULT_CATALOG_DIR)) {
+                copyDir(DEFAULT_CATALOG_DIR, catalogDir);
+            }
+            else {
+                addTpchCatalog(catalogDir);
+            }
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to add connectors", e);
+        }
+    }
+
+    private static void addTpchCatalog(Path catalogDir)
+            throws IOException
+    {
+        if (!isDirectory(catalogDir)) {
+            createDirectories(catalogDir);
+            write(catalogDir.resolve("tpch.properties"), "connector.name=tpch".getBytes());
+        }
+    }
+
+    private static void addWorkerConfig(Path configDir)
```

Comment:
It'll be nice if there's a separate method which adds all the hard-coded files needed. And also makes it clear that the hard-coded configs are for the worker.